pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -94,12 +94,16 @@ const (
 	MaxTotalScore int64 = math.MaxInt64
 )
 
-// Status indicates the result of running a plugin. It consists of a code and a
-// message. When the status code is not `Success`, the reasons should explain why.
+// Status indicates the result of running a plugin. It consists of a code, a
+// message and extra information for the status. When the status code is not `Success`,
+// the reasons should explain why, and extraInfos could provide supplementary information
+// for the reasons, e.g. extraInfos of status for FilterPlugin could provide specific
+// PredicateFailureReason which could be used by other components(e.g. kubelet).
 // NOTE: A nil Status is also considered as Success.
 type Status struct {
-	code    Code
-	reasons []string
+	code       Code
+	reasons    []string
+	extraInfos []interface{}
```

Comment:
As said in https://github.com/kubernetes/kubernetes/issues/86803#issuecomment-573513174, add `extranInfos` to provide extra info for Status. Howerver it is only needed for kubelet now.  Any comment/suggestion for it? 

cc @ahg-g @Huang-Wei

Reply:
I can't think of a good way without modifying the struct either. Another option is to make reasons as `[]interface{}`, so that we can adapt to the simple case (string) as well as complex case (InsufficientResource reason).